resource/SQLiteDB.js

```diff
@@ -0,0 +1,49 @@
+    // Please look at the README for debugging to the console
+    const Testdb = new sqliteDatabase('airlinesTest.db', {memory: true});
+    Testdb.prepare('CREATE TABLE `tbl_airports` (`code` char(3) NOT NULL,`name` varchar(45) NOT NULL,PRIMARY KEY (`code`),UNIQUE (`code`));').run();
+
+    Testdb.prepare('CREATE TABLE `tbl_flights` (`id` INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,`departure` char(3) NOT NULL,`destination` char(3) ' +
+        'NOT NULL,`departure_date` TEXT NOT NULL,`arrival_date` TEXT NOT NULL,`flight_number` int(11) NOT NULL,UNIQUE (`id`), FOREIGN KEY (`departure`) ' +
+        'REFERENCES `tbl_airports` (`code`), FOREIGN KEY (`destination`) REFERENCES `tbl_airports` (`code`),CONSTRAINT `departure!=destination` CHECK (`departure` != `destination`),' +
+        'CONSTRAINT `departure_date<destination_date` CHECK (`departure_date` < `arrival_date`));').run();
+    // create index for `destination` for tbl_flights
+    Testdb.prepare('CREATE UNIQUE INDEX destination_idx ON tbl_flights (destination);').run();
+    // create index for `departure` for tbl_flights
+    Testdb.prepare('CREATE UNIQUE INDEX departure_idx ON tbl_flights (departure);').run();
+
+    Testdb.prepare('CREATE TABLE `tbl_users` (`id` INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,`username` varchar(45),`password` varchar(255),`display_name` varchar(45) DEFAULT NULL,`role` int(11) ' +
+        'NOT NULL,`email` varchar(45),`phone` varchar(15),UNIQUE (`id`),CONSTRAINT `identifier_exists` CHECK ((`username` IS NOT NULL) OR (`email` IS NOT NULL) OR (`phone` IS NOT NULL)));').run();
+
+    Testdb.prepare('CREATE TABLE `tbl_tickets` (`flight` int(11) NOT NULL,`seat_row` int(11) NOT NULL,`seat` char(1) NOT NULL,`class` int(11) NOT NULL,`reserver` int(11) DEFAULT NULL,`price` int(11) ' +
+        'DEFAULT NULL,`reservation_timeout` datetime DEFAULT NULL,`booking_id` VARCHAR(45) DEFAULT NULL,PRIMARY KEY (`flight`,`seat_row`,`seat`),CONSTRAINT `flight` FOREIGN KEY (`flight`) REFERENCES `tbl_flights` (`id`),' +
+        'CONSTRAINT `reserver` FOREIGN KEY (`reserver`) REFERENCES `tbl_users` (`id`),CONSTRAINT `reserved_check` CHECK ( (`reserver` IS NULL AND `price` IS NULL AND `reservation_timeout` IS NULL)OR (`reserver` IS NOT NULL ' +
+        'AND `price` IS NOT NULL AND `reservation_timeout` IS NULL)OR (`reserver` IS NOT NULL AND `price` IS NULL AND `reservation_timeout` IS NOT NULL) ),CONSTRAINT `booking_id_check` CHECK ((`reserver` IS NULL AND `booking_id` IS NULL) ' +
+        'OR (`reserver` IS NOT NULL AND `booking_id` IS NOT NULL)));').run();
+    // create index for `reserver` for tbl_tickets
+    Testdb.prepare('CREATE UNIQUE INDEX reserver_idx ON tbl_tickets (reserver);').run();
+
+    Testdb.prepare('INSERT INTO `tbl_users` (`username`, `role`) VALUES ("user", "1");').run();
+
+    Testdb.prepare('INSERT INTO `tbl_airports` (`code`, `name`) VALUES ' +
+        '("DSA", "test"),("FSE", "test2"), ("TWS", "test3"),' +
```

Comment:
And where it's more than one insertion, save and reuse the prepared statement instead of packing multiple rows into one call.

Reply:
What is so bad about having it be in one line?

Reply:
If you're using the prepared-statement feature, as I suggested [here](https://github.com/utopia-airlines/UtopiaSearchService/pull/6/files#r294101330), this will turn into `.run('DS', 'test', 'FSE', 'test2', 'TWS', 'test3', 'TQI', 'test4', 'TUB', 'test5', 'CAR', 'test6')`. With just two parameters per call, and one of fixed length, it is just *barely* possible to keep track of which is which and tell by sight that you're not missing any. For any of the more complicated ones, you'd have to resort to whitespace, which while a good start isn't sufficient (because automated reformattings are a common debugging tool, if nothing else), to maintain even minimal readability.

There's also the point that (again if you're using the prepared-statement feature properly) the number of parameters in the query string has to match the number of parameters passed, which is easy to do if it's the number of columns you're inserting into one row, but less easy to verify if you've got three or five or eight rows in one call.

Reply:
Yes, you are correct. However in #7, I have moved the sql commands to a separate file.